src/Drupal/DrupalExtension/Context/RawDrupalContext.php

```diff
@@ -524,7 +580,14 @@ public function loggedIn() {
 
     // As a last resort, if a logout link is found, we are logged in. While not
     // perfect, this is how Drupal SimpleTests currently work as well.
-    return $page->findLink($this->getDrupalText('log_out'));
+    if ($page->findLink($this->getDrupalText('log_out'))) {
+      return TRUE;
+    }
+
+    // The user appears to be anonymous. Clear the current user from the user
+    // manager so this reflects the actual situation.
+    $this->getUserManager()->setCurrentUser(FALSE);
+    return FALSE;
```

Comment:
This addition should help to keep the current user in the context in sync with the actual state in the website. It is always possible the user has been logged out without by performing certain actions in the website - without the context having knowledge of this fact.

Now, if we detect the user is actually logged out in the site, the current user is updated accordingly.